thundra/application/global_application_info_provider.py

```diff
@@ -39,3 +42,12 @@ def get_application_info_from_config():
     def update(self, opts):
         filtered_opts = {k: v for k, v in opts.items() if v is not None}
         self.application_info.update(filtered_opts)
+
+        if not self.application_info.get('applicationInstanceId'):
```

Comment:
Instead of doing these in the update method, wouldn't it be better if we decide what would be the `applicationId` and `applicationInstanceId` before calling the update method outside and pass this values as `opts`?